src/benchmarks/micro/libraries/System.Collections/PriorityQueue/Perf_PriorityQueue.cs

```diff
@@ -0,0 +1,110 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Extensions;
+using MicroBenchmarks;
+
+namespace System.Collections.Tests
+{
+    [BenchmarkCategory(Categories.Libraries, Categories.Collections, Categories.GenericCollections)]
+    [GenericTypeArguments(typeof(int), typeof(int))]
+    [GenericTypeArguments(typeof(string), typeof(string))]
+    [GenericTypeArguments(typeof(Guid), typeof(Guid))]
+    public class Perf_PriorityQueue<TElement, TPriority>
+    {
+        [Params(10, 100, 1000)]
+        public int Size;
+
+        private (TElement Element, TPriority Priority)[] _items;
+        private PriorityQueue<TElement, TPriority> _priorityQueue;
+        private PriorityQueue<TElement, TPriority> _prePopulatedPriorityQueue;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            _items = ValuesGenerator.Array<TElement>(Size).Zip(ValuesGenerator.Array<TPriority>(Size)).ToArray();
```

Comment:
`ValuesGenerator` always generates the same data and here `TElement` is always the same as `TPriority` so for each `(TElement Element, TPriority Priority)` `Element` and `Priority` are going to be equal. Is that OK?

If not we could do sth like this:

```suggestion
            var elements = ValuesGenerator.Array<TElement>(Size * 2);
            var priorities = ValuesGenerator.Array<TPriority>(Size * 2);
            _items = elements.Take(Size).Zip(priorities.Skip(Size)).ToArray();
```

Reply:
It shouldn't matter in this case, since `priorities` is what really drives the heap formation. Alternatively I was thinking we could update `ValuesGenerator` to also include support for tuple generation.

Reply:
Is performance going to be impacted measurably based on whether all priorities are different or many priorities are the same?

Reply:
If many priorities are the same then I expect that the average depth of a siftup/siftdown operation would be shortened. If all priorities were the same then enqueue and dequeue becomes _O(1)_ (since the heap property is always satisfied).